### ISSUE TITLE
.defaults(options) does not inherit options

### ISSUE THREAD

**Issue by gr2m** [pouchdb/pouchdb#5922]
### Issue

We use `PouchDB.defaults(options)` extensively. I would expect that if I do something like this:

```js
var MyPouchDB = PouchDB.defaults({foo: 'bar'})
var MyNewPouchDB = MyPouchDB.defaults({baz: 'qux'})
```

then both options (`{foo, baz}`) would be passed by the `MyNewPouchDB` constructor implicitly, but this does not seem to be the case. Is this by design? If this is a bug, I’d be happy to look into it

### Info
- Environment: tested in latest Chrome 54 on pouchdb.com
- Platform: Chrome

### Reproduce
This tries to send a request to `https://example.com/db` with the `x-myheader` header, as expected 👍 

```js
MyPouchDB = PouchDB.defaults({
  ajax: {
    headers: {
      'x-myheader': 'foobarbaz'
    }
  }
})
var db = new MyPouchDB('db')
db.replicate.to('https://example.com/db')
```

But this does not (note the additional `.defaults({})`

```js
MyPouchDB = PouchDB.defaults({
  ajax: {
    headers: {
      'x-myheader': 'foobarbaz'
    }
  }
}).defaults({})
var db = new MyPouchDB('db')
db.replicate.to('https://example.com/db')
```

**Comment by nolanlawson** [comment]
The `defaults` should be transitive. This looks like a bug!

**Comment by gr2m** [comment]
okay I’m looking into it now. Can we make this part of v6.0.8 if I get it done before December?

**Comment by nolanlawson** [comment]
Fixed in #5953
